src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c

```diff
@@ -446,31 +472,68 @@ krb5_encode_krbsecretkey(krb5_key_data *key_data_in, int n_key_data,
         }
     }
 
+    ret = k5alloc(sizeof(struct berval), &err);
+    if (ret == NULL)
+        goto cleanup;
+
+    err = asn1_encode_sequence_of_keys(key_data, n_key_data, mkvno, &code);
+    if (err)
+        goto cleanup;
+
+    ret->bv_len = code->length;
+    ret->bv_val = code->data;
+
+    *ret_out = ret;
+    ret = NULL;
+
+cleanup:
+    free(code);
```

Comment:
if `asn1_encode_sequence_of_keys` ever returns an error, can `code` still be allocated memory?